crates/runtime/src/dataconnector/s3.rs

```diff
@@ -161,8 +161,14 @@ impl DataConnectorFactory for S3Factory {
                 validator.validate(&mut params).await?;
             }
 
-            // Initialize the AWS SDK and make it available.
-            let _ = object_store_aws_sdk::initialize_sdk_config().await;
+            // `initialize_sdk_config` emits a warning if the credentials provider cannot be initialized
+            // so we skip it if the auth method is public.
+            match params.parameters.get("auth").expose().ok() {
```

Comment:
I'm under impression is that the same could also happen for `auth: key` where credentials are explicitly provided, we might just want to eliminate tracing for `aws_config::profile::credentials:` but it can be usefull so I'm keeping as a quick workaround for s3 public and we might want to revisit this logic